graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsDataLoaderProvider.kt

```diff
@@ -87,50 +85,53 @@ class DgsDataLoaderProvider(
     }
 
     private fun addDataLoaderFields() {
-        applicationContext.getBeansWithAnnotation(DgsComponent::class.java).values.forEach { dgsComponent ->
-            val javaClass = AopUtils.getTargetClass(dgsComponent)
+        val dataLoaders = applicationContext.getBeansWithAnnotation(DgsComponent::class.java)
+        dataLoaders.values.distinctBy { AopUtils.getTargetClass(it) }
```

Comment:
Why doing distinctBy here, because that would avoid/hide the problem? It should just fail if multiple with the same name are found I think

Reply:
So I added a comment earlier as well `This was needed because I was seeing multiple beans (generic and root) for the same class in the context of tests.`

What I was seeing is that in the context of tests, I see 2 beans for the same provider class - one was a Generic bean and the other a Root bean. I'm not sure if this is expected behavior though.